### ISSUE TITLE
No sample file generated

### ISSUE THREAD

**Issue by frnmz** [SMByC/AcATaMa#10]
Hi! Thanks for your plugin!
But i found bugs that the plugin were not generated any sample file.
When I click "Generate the sampling" it says :
```
2021-09-12T07:05:26     INFO    AcATaMa : Could not load layer: /mnt/65625765-a7a4-482f-8971-5e97fb483b25/linux/documents/Kerja/mba nurul pandeglang/Citra Pandeglang/sentinel/mencoba/aaaaaa
2021-09-12T07:05:26     SUCCESS    AcATaMa : Generate the simple random sampling, completed
```
Thus no sample file were generated.
I've tried both geopackage and shapefile, but still have the same issues.
I've tried reinstalling the plugin too.
I'm using QGIS:
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
</head><body>

QGIS version | 3.20.2-Odense | QGIS code branch | Release 3.20
-- | -- | -- | --
Qt version | 5.15.2
Python version | 3.9.6
GDAL/OGR version | 3.2.2
PROJ version | 7.2.1
EPSG Registry database version | v10.008 (2020-12-16)
GEOS version | 3.9.0-CAPI-1.16.2
SQLite version | 3.34.1
PDAL version | 2.2.0
PostgreSQL client version | 13.3
SpatiaLite version | 5.0.1
QWT version | 6.1.5
QScintilla2 version | 2.11.5
OS version | Fedora 34 (Workstation Edition)
  |   |   |  
Active Python plugins | temporalprofiletoolCalculateGeometryMetaSearchdb_managerprocessingSemiAutomaticClassificationPluginAcATaMa

</body></html>

Here's a gif I hope it can help you:
![ezgif-6-de7a4dc38925](https://user-images.githubusercontent.com/18339774/132966589-a32849ba-3684-4ae3-aebb-df9c73b4ae60.gif)

**Comment by XavierCLL** [comment]
Hi @frnmz 

Thanks for your detailed report!

I cannot reproduce your issue, I'm almost certain that the problem is because you don't have write permission in that path, try saving that file in e.g. your home directory.

Regards,
Xavier

**Comment by frnmz** [comment]
Hi thanks for your reply!
I dont think its permission issue because I've tried saving the sample file in home and another folder and still encounter the issue.
There is this python warning too that might be helpful:
```
2021-09-12T08:17:25     WARNING    warning:/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/about_dialog.py:37: ResourceWarning: unclosed file 
              html_text = open(about_file).read()
             
             traceback: File "", line 1, in 
              File "/usr/lib64/python3.9/site-packages/qgis/utils.py", line 355, in startPlugin
              if not _startPlugin(packageName):
              File "/usr/lib64/python3.9/site-packages/qgis/utils.py", line 335, in _startPlugin
              plugins[packageName] = package.classFactory(iface)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/__init__.py", line 66, in classFactory
              return AcATaMa(iface)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/acatama.py", line 72, in __init__
              self.about_dialog = AboutDialog()
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/about_dialog.py", line 37, in __init__
              html_text = open(about_file).read()
             
2021-09-12T08:18:01     WARNING    warning:/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/about_dialog.py:37: ResourceWarning: unclosed file 
              html_text = open(about_file).read()
             
             traceback: File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/acatama.py", line 127, in run
              self.dockwidget = AcATaMaDockWidget()
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 68, in __init__
              self.setup_gui()
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 80, in setup_gui
              self.about_dialog = AboutDialog()
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/about_dialog.py", line 37, in __init__
              html_text = open(about_file).read()
             
2021-09-12T08:18:03     WARNING    warning:/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py:266: DeprecationWarning: an integer is required (got type float). Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
              self.nodata_ThematicRaster.setValue(get_nodata_value(layer))
             
             traceback: File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 266, in select_thematic_raster
              self.nodata_ThematicRaster.setValue(get_nodata_value(layer))
             
2021-09-12T08:18:10     WARNING    warning:/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/sampling.py:280: DeprecationWarning: QgsVectorFileWriter constructor is deprecated
              writer = QgsVectorFileWriter(self.output_file, "System", fields, QgsWkbTypes.Point, thematic_CRS, file_format)
             
             traceback: File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 142, in 
              self.widget_generate_SimpRS.QPBtn_GenerateSampling.clicked.connect(lambda: do_simple_random_sampling(self))
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/utils/system_utils.py", line 36, in wrapper
              return func(*args, **kwargs)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/sampling.py", line 102, in do_simple_random_sampling
              sampling.generate_sampling_points(pixel_values, number_of_samples, min_distance,
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/utils/system_utils.py", line 36, in wrapper
              return func(*args, **kwargs)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/utils/system_utils.py", line 86, in wrapper
              obj_returned = func(*args, **kwargs)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/sampling.py", line 280, in generate_sampling_points
              writer = QgsVectorFileWriter(self.output_file, "System", fields, QgsWkbTypes.Point, thematic_CRS, file_format)
             
2021-09-12T08:18:10     WARNING    warning:/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/sampling.py:318: DeprecationWarning: QgsSpatialIndex.insertFeature() is deprecated
              self.index.insertFeature(f)
             
             traceback: File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 142, in 
              self.widget_generate_SimpRS.QPBtn_GenerateSampling.clicked.connect(lambda: do_simple_random_sampling(self))
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/utils/system_utils.py", line 36, in wrapper
              return func(*args, **kwargs)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/sampling.py", line 102, in do_simple_random_sampling
              sampling.generate_sampling_points(pixel_values, number_of_samples, min_distance,
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/utils/system_utils.py", line 36, in wrapper
              return func(*args, **kwargs)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/utils/system_utils.py", line 86, in wrapper
              obj_returned = func(*args, **kwargs)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/sampling.py", line 318, in generate_sampling_points
              self.index.insertFeature(f)
```

**Comment by XavierCLL** [comment]
Ok, then I'll investigate, could you please start Qgis from the terminal do the same test and copy what the terminal says? thanks

**Comment by frnmz** [comment]
This is what it says:
```
[lam@gamla-stan ~]$ qgis
Warning: QSocketNotifier: Can only be used with threads started with QThread
Warning: QFont::setPointSize: Point size <= 0 (-1), must be greater than 0
Warning: QFont::setPointSize: Point size <= 0 (-1), must be greater than 0
** INITIALIZING AcATaMa
Warning: setGrabPopup called with a parent, QtWaylandClient::QWaylandXdgSurface(0x557da5bf49c0) which does not match the current topmost grabbing popup, QtWaylandClient::QWaylandXdgSurface(0x557dab617a00) According to the xdg-shell protocol, this is not allowed. The wayland QPA plugin is currently handling it by setting the parent to the topmost grabbing popup. Note, however, that this may cause positioning errors and popups closing unxpectedly because xdg-shell mandate that child popups close before parents
Warning: Wayland does not support QWindow::requestActivate()
Warning: Wayland does not support QWindow::requestActivate()
Warning: Wayland does not support QWindow::requestActivate()
Warning: Wayland does not support QWindow::requestActivate()
Warning: Wayland does not support QWindow::requestActivate()
Warning: Wayland does not support QWindow::requestActivate()
ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

ERROR 10: Pointer 'hLayer' is NULL in 'OGR_L_GetLayerDefn'.

ERROR 10: Pointer 'hDefn' is NULL in 'OGR_F_Create'.

ERROR 10: Pointer 'hFeat' is NULL in 'OGR_F_SetGeometryDirectly'.

** UNLOAD AcATaMa
```

**Comment by XavierCLL** [comment]
mmm weird, have you tried with another thematic file? (only if it's possible) could you share with me your thematic file for doing a local test?

**Comment by frnmz** [comment]
yupp i've tried with another file.
sure! here is my file:
[klasifikasi 2019 final.zip](https://github.com/SMByC/AcATaMa/files/7148855/klasifikasi.2019.final.zip)

**Comment by XavierCLL** [comment]
Your thematic file works perfect in my system with the same Qgis and Acatama version. Your problem is related to some issue between Qgis and Gdal (or some library that use gdal), have you installed Qgis and Gdal from the official Fedora repository? are you using and unofficial repository? or did you build from source?

Try for example a clean environment for that, installing Conda and Qgis from there: https://anaconda.org/conda-forge/qgis and init the Qgis inside the conda instance.

**Comment by XavierCLL** [comment]
could you follow this tutorial from the Qgis terminal without error? https://opensourceoptions.com/blog/pyqgis-create-a-shapefile/

**Comment by frnmz** [comment]
> could you follow this tutorial from the Qgis python terminal without getting any error? https://opensourceoptions.com/blog/pyqgis-create-a-shapefile/

i got this error:
`2021-09-12T09:41:41     CRITICAL    Layer is not valid : The layer ~/newpoints.shp is not a valid layer and can not be added to the map. Reason:  `

`2021-09-12T09:41:41     WARNING    Cannot open ~/newpoints.shp.()`

```
2021-09-12T09:41:27     WARNING    warning:/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/about_dialog.py:37: ResourceWarning: unclosed file 
              html_text = open(about_file).read()
             
             traceback: File "", line 1, in 
              File "/usr/lib64/python3.9/site-packages/qgis/utils.py", line 355, in startPlugin
              if not _startPlugin(packageName):
              File "/usr/lib64/python3.9/site-packages/qgis/utils.py", line 335, in _startPlugin
              plugins[packageName] = package.classFactory(iface)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/__init__.py", line 66, in classFactory
              return AcATaMa(iface)
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/acatama.py", line 72, in __init__
              self.about_dialog = AboutDialog()
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/about_dialog.py", line 37, in __init__
              html_text = open(about_file).read()
             
2021-09-12T09:41:41     WARNING    warning::1: DeprecationWarning: QgsVectorFileWriter constructor is deprecated
             
             traceback: File "/usr/share/qgis/python/console/console_sci.py", line 355, in keyPressEvent
              self.paste()
              File "/usr/share/qgis/python/console/console_sci.py", line 466, in paste
              self.insertFromDropPaste(stringPaste)
              File "/usr/share/qgis/python/console/console_sci.py", line 486, in insertFromDropPaste
              self.runCommand(self.text())
              File "/usr/share/qgis/python/console/console_sci.py", line 526, in runCommand
              more = self.runsource(src)
              File "/usr/share/qgis/python/console/console_sci.py", line 559, in runsource
              return super(ShellScintilla, self).runsource(source, filename, symbol)
              File "/usr/lib64/python3.9/code.py", line 74, in runsource
              self.runcode(code)
              File "/usr/lib64/python3.9/code.py", line 90, in runcode
              exec(code, self.locals)
              File "", line 1, in 
```

the error come after this line:
`layer = iface.addVectorLayer(fn, '', 'ogr')`

**Comment by frnmz** [comment]
> Your thematic file works perfect in my system with the same Qgis and Acatama version. Your problem is related to some issue between Qgis and Gdal (or some library that use gdal), have you installed Qgis and Gdal from the official Fedora repository? are you using and unofficial repository? or did you build from source?

 I'm using the QGIS official fedora repo that are in their webs:
 
![image](https://user-images.githubusercontent.com/18339774/132969747-f257b81a-e2e3-4959-b952-935836bc0958.png)

> 
> Try for example a clean environment for that, installing Conda and Qgis from there: https://anaconda.org/conda-forge/qgis and init the Qgis inside the conda instance.

I'm trying right now, will report after this

**Comment by frnmz** [comment]
> Try for example a clean environment for that, installing Conda and Qgis from there: https://anaconda.org/conda-forge/qgis and init the Qgis inside the conda instance.

Can confirm Acatama works well in QGIS that runs in a new conda environtment.
![image](https://user-images.githubusercontent.com/18339774/132970037-9fde1482-0478-4e98-9c70-5ccc740a4157.png)
So is the problem in my default qgis python environment? If yes, how do i create one from default?

**Comment by XavierCLL** [comment]
Good to know that finally it works, so the original error was in a Qgis function that I use in Acatama, that function in the background use Gdal, then your problem is either Qgis or Gdal (or python). First, try to reinstall all both apps and some deps like python3-gdal. Second check all your repos installed in fedora (update your system), or maybe there is a repository generating a conflict or problem, is your Fedora 34 updating from an old Fedora version? (I use Arch and I don't have much experience with Fedora, sorry). If that doesn't work try to use another (unofficial repo that provides Qgis) or just use that Qgis from conda that you have installed.

**Comment by frnmz** [comment]
I just tried `dnf reinstall python3-gdal` and now Acatama works!
Maybe my problem and the resolve (to reinstall gdal) can be useful if you were have a troubleshooting docs for this plugin.
Thanks for your guidance and nice plugin!

**Comment by XavierCLL** [comment]
Hi @frnmz 

Thanks for the follow-up of this issue, I've never used both plugins at the same time, Acatama doesn't need special dependencies, only gdal and python-gdal and both always come with Qgis, but maybe SAC are using or installing other versions and they could be in conflict. 

Definitely, it requires more research and testing, I'll do it when I have time for it, please update this chat with any feedback if you find something related.

Thanks, Xavier

**Comment by XavierCLL** [comment]
Hi @frnmz, If you have the chance to try the new version will be great, the latest version has a lot of changes and maybe fix this issue. On the other hand, I'm going to close this because this issue is not entirely related to this plugin, it is a collateral effect with other plugins installed, but the discussion can continue of course.

Thanks again. 
Xavier

**Comment by frnmz** [comment]
Hi @XavierCLL, I just tried using the latest version available in QGIS plugin repo (22.10.23), what I have is this error when generating the sample:

> Traceback (most recent call last):
  File "/var/home/farhanmakarim/.var/app/org.qgis.qgis/data/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/utils/system_utils.py", line 36, in wrapper
    return func(*args, **kwargs)
  File "/var/home/farhanmakarim/.var/app/org.qgis.qgis/data/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/sampling_design.py", line 137, in do_simple_random_sampling
    ResponseDesign(sampling_layer_generated)
  File "/var/home/farhanmakarim/.var/app/org.qgis.qgis/data/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/response_design.py", line 45, in __init__
    self.points = self.get_points_from_shapefile()
  File "/var/home/farhanmakarim/.var/app/org.qgis.qgis/data/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/core/response_design.py", line 111, in get_points_from_shapefile
    for enum_id, qgs_feature in enumerate(self.sampling_layer.getFeatures(), start=1):
AttributeError: 'QgsRasterLayer' object has no attribute 'getFeatures'

I'm on Fedora 37 Silverblue with latest Flatpak QGIS and SAC plugin installed.

But! It works fine when I'm using it in QGIS inside a podman container installed from fedora repo. Even with SAC plugin installed.
Maybe QGIS flatpak's python environment is the one messing the plugin...

Also thanks for still keeping this issue thread alive...

**Comment by XavierCLL** [comment]
Hi @frnmz thanks for your reply and tests,

The latest error message that you have received is new, it seems that the plugin is trying to retrieve sampling features from a raster file, which is not possible. The file format that the plugin requires for the sampling file is either a GeoPackage file (.gpkg) or a Shape file (.shp), both of which are vector file formats.

Therefore, when you click on "generate the sampling," you should make sure to save the sampling file in either of these formats.

Could you please check that, Thanks!
